app/services/CategorisationService.scala

```diff
@@ -39,28 +43,68 @@ class CategorisationService @Inject() (
     recordId: String,
     longerCode: Boolean = false
   )(implicit hc: HeaderCarrier): Future[CategorisationInfo] =
-    ottConnector
-      .getCategorisationInfo(
-        commodityCode,
-        request.eori,
-        request.affinityGroup,
-        Some(recordId),
-        country,
-        LocalDate.now()
-      )
-      .flatMap { response =>
-        CategorisationInfo.build(response, commodityCode, longerCode) match {
-          case Some(categorisationInfo) => Future.successful(categorisationInfo)
-          case _                        =>
-            Future.failed(new RuntimeException("Could not build categorisation info"))
+    profileConnector.getTraderProfile(request.eori).flatMap { profile =>
+      ottConnector
+        .getCategorisationInfo(
+          commodityCode,
+          request.eori,
+          request.affinityGroup,
+          Some(recordId),
+          country,
+          LocalDate.now()
+        )
+        .flatMap { response =>
+          CategorisationInfo.build(response, commodityCode, profile, longerCode) match {
+            case Some(categorisationInfo) =>
+              Future.successful(categorisationInfo)
+
+            case None =>
+              logger.error("Could not build categorisation info")
+              Future.failed(new RuntimeException("Could not build categorisation info"))
+          }
         }
-      }
+    }
 
   def calculateResult(
     categorisationInfo: CategorisationInfo,
     userAnswers: UserAnswers,
     recordId: String
-  ): Scenario =
+  ): Scenario = {
+    val category1Assessments = categorisationInfo.categoryAssessments.filter(ass => ass.isCategory1)
+
+    val hasNiphlAssessments = category1Assessments.exists(ass => ass.isNiphlsAnswer)
+
+    val category1AssessmentsWithoutNiphl = category1Assessments.filter(ass => !ass.isNiphlsAnswer)
+
+    val listOfAnswers = categorisationInfo.getAnswersForQuestions(userAnswers, recordId)
+
+    val areThereCategory1QuestionsWithNoExemption =
+      listOfAnswers.exists(x => x.answer.contains(AssessmentAnswer.NoExemption) && x.question.isCategory1)
+
+    if (categorisationInfo.isNiphlAuthorised) {
```

Comment:
I think the logic is slightly topsy-turvy - you only care about isNiphlAuthorised if its a niphls assessment

I think this is the same?

`    if (isNiphlsAssessment) {
      if (!categorisationInfo.isNiphlAuthorised || areThereCategory1QuestionsWithNoExemption) {
          Category1Scenario
        } else {
          Category2Scenario
        }
    } else {
      calculateResultWithoutNiphl(categorisationInfo, userAnswers, recordId)
    }`
    
    A test fails but the test data is wrong

Reply:
Logic updated and fix failing tests